src/Umbraco.PublishedCache.HybridCache/Persistence/ContentSourceDto.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Umbraco.Cms.Core.Models;
+
+namespace Umbraco.Cms.Infrastructure.HybridCache.Persistence
+{
+    // read-only dto
+    internal class ContentSourceDto : IReadOnlyContentBase
+    {
+        public int Id { get; set; }
```

Comment:
Hey if this is really read only shouldn't the setters then be removed?

For example with a constructor:
```cs
public int Id { get; }
```

or

```cs
public required int Id { get; init; }
```

If it has to be set on creation. (The `required` keyword is optional and just means it has to be set to a value)

I would also imagine that you would want this for the classes that uses `[ImmutableObject(true)]`?

Just some a thought I had when looking it over 😄

Reply:
That's a good point, I've updated it now 👍 Thanks